### ISSUE TITLE
[Feature] Info about when invoices or quotes were sent to clients

### ISSUE THREAD

**Issue by mheiduk** [InvoicePlane/InvoicePlane#1025]
My client is missing an info about when invoices or quotes were sent to clients. Especially when clients are missing the due date it would be good to know when they were getting a reminder.

## Expected Behavior
A date column/field "Sent On" for invoices and quotes.

## Current Behavior
No info when invoices or quotes were sent.

## Possible Solution/Implementation/Description
New columns in the db tables `invoice_date_sent DATETIME` and `quote_date_sent DATETIME` with the info about when invoices were sent to the clients. Will be updated in the `mark_sent` functions.

New column on the invoices/quotes table overview. Furthermore a new datetime field at invoice/quote detail level.

A pull request is already in the making, but before that I have to test it a little bit more. Any further input/suggestion is welcome.

**Comment by nielsdrost7** [comment]
> Any further input/suggestion is welcome
Well, I love the idea.
Are you sure you don't want a list of all the dates you sent that invoice/quote to the customer?
That means another table instead of those new fields.

> `invoice_date_sent`
I would already start with the 'laravel' type dates, example: `invoice_sent_at`

Those new date fields, I wouldn't place them at the end of the table, but `after` a certain column.

Just a suggestion, otherwise it's going to be a mess with all kinds of fields spread everywhere in those 2 tables.

**Comment by mheiduk** [comment]
> > Any further input/suggestion is welcome
> > Well, I love the idea.
> > Are you sure you don't want a list of all the dates you sent that invoice/quote to the customer?
> > That means another table instead of those new fields.
> 
> > `invoice_date_sent`
> > I would already start with the 'laravel' type dates, example: `invoice_sent_at`
> 
> Those new date fields, I wouldn't place them at the end of the table, but `after` a certain column.
> 
> Just a suggestion, otherwise it's going to be a mess with all kinds of fields spread everywhere in those 2 tables.

Yes, in my current draft I add the new columns after `invoice_date_due` respectively `quote_date_expires`.

**Comment by Verony-makesIT** [comment]
@mheiduk, 
my function "**update_invoice_due_dates($invoice_id) in Mdl_invoices.php**" already does this **for invoices**. 

See Feature Request #895.
When sending an invoice, the fields (invoice_date_created and invoice_date_due) in the ip_invoices db are updated to the current dates. 
That way you can check which date the invoice was actually sent and which is the due date.
However, this behavior can be adjusted in "Settings - Invoices".
![image](https://github.com/InvoicePlane/InvoicePlane/assets/30659226/fc2808f4-9554-4792-b854-0620163c31f5)

**Comment by mheiduk** [comment]
@Verony-makesIT thanks! I am aware of this feature, but this led to confusion for my client, because the invoice creation date will also be updated. This is in my opinion a wrong behavior, the creation date should always be the same.

If someone wants this feature, it should be at least opt-in.

**Comment by Verony-makesIT** [comment]
@mheiduk, 
it can indeed be confusing, and strictly speaking, you and/or your client are right. 
Those who have to pay the invoice (before the due date) don't really care when the invoice was created, originally or technically. What does matter to them is that they can use the set days until the due date to pay their invoices. 
Now, by overwriting the creation date with the shipment date, the due date is set up correctly. 
In my opinion, both for the supplier and customer, this is the most fair and equitable approach.
_Just my 2 cents..._

And if your customer is not in favor of this, they can easily change this setting.
